### ISSUE TITLE
Internal server error when running docker

### ISSUE THREAD

**Issue by pandoraxcc** [docker/for-mac#7362]
### Description

Hello, encountered the error with docker after installing the latest (as for now update) Docker Desktop 4.32.0 (157355).

After running the app, on the **build** tab it says "Internal Server Error".
Prior to update, had some issues accessing the list of the builds as well.

<img width="1432" alt="docker" src="https://github.com/user-attachments/assets/2c91ceff-e86c-46d1-b3e8-d6cc95abf99e">

### Reproduce

1. open installed docker app on Mac OS, 2. switch to build tab 

### Expected behavior

Docker should load the interface with history of previous and current builds

### docker version

```bash
{
  "ApiVersion": "1.46",
  "Arch": "amd64",
  "BuildTime": "2024-06-29T00:02:50.000000000+00:00",
  "Components": [
    {
      "Details": {
        "ApiVersion": "1.46",
        "Arch": "amd64",
        "BuildTime": "2024-06-29T00:02:50.000000000+00:00",
        "Experimental": "false",
        "GitCommit": "662f78c",
        "GoVersion": "go1.21.11",
        "KernelVersion": "6.6.32-linuxkit",
        "MinAPIVersion": "1.24",
        "Os": "linux"
      },
      "Name": "Engine",
      "Version": "27.0.3"
    },
    {
      "Details": {
        "GitCommit": "ae71819c4f5e67bb4d5ae76a6b735f29cc25774e"
      },
      "Name": "containerd",
      "Version": "1.7.18"
    },
    {
      "Details": {
        "GitCommit": "v1.1.13-0-g58aa920"
      },
      "Name": "runc",
      "Version": "1.7.18"
    },
    {
      "Details": {
        "GitCommit": "de40ad0"
      },
      "Name": "docker-init",
      "Version": "0.19.0"
    }
  ],
  "GitCommit": "662f78c",
  "GoVersion": "go1.21.11",
  "KernelVersion": "6.6.32-linuxkit",
  "MinAPIVersion": "1.24",
  "Os": "linux",
  "Platform": {
    "Name": "Docker Engine - Community"
  },
  "Version": "27.0.3"
}
```

### docker info

```bash
{
  "Architecture": "x86_64",
  "BridgeNfIp6tables": true,
  "BridgeNfIptables": true,
  "CDISpecDirs": [],
  "CPUSet": true,
  "CPUShares": true,
  "CgroupDriver": "cgroupfs",
  "CgroupVersion": "2",
  "ContainerdCommit": {
    "Expected": "ae71819c4f5e67bb4d5ae76a6b735f29cc25774e",
    "ID": "ae71819c4f5e67bb4d5ae76a6b735f29cc25774e"
  },
  "Containers": 0,
  "ContainersPaused": 0,
  "ContainersRunning": 0,
  "ContainersStopped": 0,
  "CpuCfsPeriod": true,
  "CpuCfsQuota": true,
  "Debug": false,
  "DefaultRuntime": "runc",
  "DockerRootDir": "/var/lib/docker",
  "Driver": "overlay2",
  "DriverStatus": [
    [
      "Backing Filesystem",
      "extfs"
    ],
    [
      "Supports d_type",
      "true"
    ],
    [
      "Using metacopy",
      "false"
    ],
    [
      "Native Overlay Diff",
      "true"
    ],
    [
      "userxattr",
      "false"
    ]
  ],
  "ExperimentalBuild": false,
  "GenericResources": null,
  "HttpProxy": "http.docker.internal:3128",
  "HttpsProxy": "http.docker.internal:3128",
  "ID": "93c9a5d0-1cca-48dc-91d5-b0d8491496f7",
  "IPv4Forwarding": true,
  "Images": 0,
  "IndexServerAddress": "https://index.docker.io/v1/",
  "InitBinary": "docker-init",
  "InitCommit": {
    "Expected": "de40ad0",
    "ID": "de40ad0"
  },
  "Isolation": "",
  "KernelVersion": "6.6.32-linuxkit",
  "Labels": [],
  "LiveRestoreEnabled": false,
  "LoggingDriver": "json-file",
  "MemTotal": 8221057024,
  "MemoryLimit": true,
  "NCPU": 8,
  "NEventsListener": 7,
  "NFd": 50,
  "NGoroutines": 62,
  "Name": "docker-desktop",
  "NoProxy": "hubproxy.docker.internal",
  "OSType": "linux",
  "OSVersion": "",
  "OomKillDisable": false,
  "OperatingSystem": "Docker Desktop",
  "PidsLimit": true,
  "Plugins": {
    "Authorization": null,
    "Log": [
      "awslogs",
      "fluentd",
      "gcplogs",
      "gelf",
      "journald",
      "json-file",
      "local",
      "splunk",
      "syslog"
    ],
    "Network": [
      "bridge",
      "host",
      "ipvlan",
      "macvlan",
      "null",
      "overlay"
    ],
    "Volume": [
      "local"
    ]
  },
  "RegistryConfig": {
    "AllowNondistributableArtifactsCIDRs": null,
    "AllowNondistributableArtifactsHostnames": null,
    "IndexConfigs": {
      "docker.io": {
        "Mirrors": [],
        "Name": "docker.io",
        "Official": true,
        "Secure": true
      },
      "hubproxy.docker.internal:5555": {
        "Mirrors": [],
        "Name": "hubproxy.docker.internal:5555",
        "Official": false,
        "Secure": false
      }
    },
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ],
    "Mirrors": null
  },
  "RuncCommit": {
    "Expected": "v1.1.13-0-g58aa920",
    "ID": "v1.1.13-0-g58aa920"
  },
  "Runtimes": {
    "io.containerd.runc.v2": {
      "path": "runc"
    },
    "runc": {
      "path": "runc"
    }
  },
  "SecurityOptions": [
    "name=seccomp,profile=unconfined",
    "name=cgroupns"
  ],
  "ServerVersion": "27.0.3",
  "SwapLimit": true,
  "Swarm": {
    "ControlAvailable": false,
    "Error": "",
    "LocalNodeState": "inactive",
    "NodeAddr": "",
    "NodeID": "",
    "RemoteManagers": null
  },
  "SystemTime": "2024-07-23T15:31:04.358885697Z",
  "Warnings": [
    "WARNING: daemon is not using the default seccomp profile"
  ]
}
```

### Diagnostics ID

4C351BF6-BDEC-4AD9-A40D-42F805C3F0EE/20240723152553

**Comment by byronrode** [comment]
<img width="1623" alt="image" src="https://github.com/user-attachments/assets/008509c0-12dd-4e09-89b1-a61a708d24aa">

I got the same, along with this error now, since the update. Docker CLI fails to run as well, with a 500 server error. Resetting doesn't solve anything except to lose all local containers.

**Comment by zuoxianglei** [comment]
I also encountered the same problem.Hope to fix it as soon as possible. Thank you

**Comment by Phoeni0x** [comment]
Same here, any workaround to circumvent this issue?

**Comment by ulas96** [comment]
Suffering from this issue for months. Any updates?

**Comment by ulas96** [comment]
I had the same issue but solved it by doing this https://github.com/docker/for-mac/issues/7144#issuecomment-2152299467.

**Comment by crazy-max** [comment]
> I had the same issue but solved it by doing this [#7144 (comment)](https://github.com/docker/for-mac/issues/7144#issuecomment-2152299467).

Don't think it needs to remove the docker config folder, just fixing permissions similar to https://github.com/docker/for-mac/issues/7356#issuecomment-2284181696 should be enough:

```
$ sudo chown -R $(id -un):$(id -gn) ~/.docker/buildx
```
